### PR TITLE
CPL added

### PR DESCRIPTION
**Pull request by anastasia-tsvetkova**
Cutoff PL parameterized via Ep added

### CODE REVIEW

**Comment by grburgess** [comment]
@anastasia-tsvetkova you will need to run the script in /scripts/ python generate_test_data.py

and move the newly created table to astromodels/data and update that table in the git

File: astromodels/functions/functions_1D/powerlaws.py [hunk 1]
```diff
@@ -343,6 +343,83 @@ def evaluate(self, x, K, piv, index, xc):
 
         return result * unit_
 
+      
+class CPL(Function1D, metaclass=FunctionMeta):
```
Comment (grburgess) [inline astromodels/functions/functions_1D/powerlaws.py:347]: The use of CPL could be confusing as there is already a cutoff power law function. Moreover, we typically spell out the function name to be clear for users.

File: astromodels/functions/functions_1D/powerlaws.py [hunk 2]
```diff
@@ -343,6 +343,83 @@ def evaluate(self, x, K, piv, index, xc):
...
+    r"""
+    description :
+
+        A power law multiplied by an exponential cutoff parametrized via Ep
+
+    latex : $ K~\frac{x}{piv}^{index}~\exp{-x/xp} $
```
Comment (grburgess) [inline astromodels/functions/functions_1D/powerlaws.py:353]: is this the correct formula?

File: astromodels/functions/functions_1D/powerlaws.py [hunk 3]
```diff
@@ -343,6 +343,83 @@ def evaluate(self, x, K, piv, index, xc):
...
+            xp_ = xp.value
+            x_ = x.value
+
+            unit_ = self.y_unit
+
+        else:
+            unit_ = 1.0
+            K_, piv_, x_, index_, xp_ = K, piv, x, index, xp
+
+        result = nb_func.cplaw_eval(x_, K_, xp_, index_, piv_)
```
Comment (grburgess) [inline astromodels/functions/functions_1D/powerlaws.py:419]: There was a new numba function ```cpp_eval``` created but it is not called here. However, there is really no need as you can write xc = xp (2+index) and pass this to ```cplaw_eval``` without the need for extra code.

File: astromodels/functions/numba_functions.py [hunk 4]
```diff
@@ -67,6 +67,20 @@ def cplaw_eval(x, K, xc, index, piv):
     return out
 
 
+@nb.njit(fastmath=True, cache=True)
+def cpl_eval(x, K, xp, index, piv):
```
Comment (grburgess) [inline astromodels/functions/numba_functions.py:71]: This function is not needed.

**Review by grburgess — CHANGES_REQUESTED** [review]
Thank you for the addition. However, I'm curious if it is not easier to compute Ep after a fit via error propagation. This parameterization introduces functional correlations.

File: astromodels/functions/numba_functions.py (hunk 4 above)
Comment (anastasia-tsvetkova) [reply astromodels/functions/numba_functions.py]: deleted

File: astromodels/functions/functions_1D/powerlaws.py (hunk 3 above)
Comment (anastasia-tsvetkova) [reply astromodels/functions/functions_1D/powerlaws.py]: fixed

File: astromodels/functions/functions_1D/powerlaws.py (hunk 1 above)
Comment (anastasia-tsvetkova) [reply astromodels/functions/functions_1D/powerlaws.py]: fixed: CPL -> Cutoff_powerlaw_Ep

**Review by grburgess — APPROVED** [review]

**Pull request merged**